torch_geometric/nn/models/revgnn.py

```diff
@@ -0,0 +1,306 @@
+                element_original.set_(element_inverted)
+
+        # Compute gradients with grad enabled
+        with torch.set_grad_enabled(True):
+            detached_inputs = []
+            for element in inputs:
+                if isinstance(element, torch.Tensor):
+                    detached_inputs.append(element.detach())
+                else:
+                    detached_inputs.append(element)
+            detached_inputs = tuple(detached_inputs)
+            for det_input, requires_grad in zip(detached_inputs,
+                                                ctx.input_requires_grad):
+                if isinstance(det_input, torch.Tensor):
+                    det_input.requires_grad = requires_grad
+            temp_output = ctx.fn(*detached_inputs)
+
+        if not isinstance(temp_output, tuple):
+            temp_output = (temp_output, )
+        filtered_detached_inputs = tuple(
+            filter(
+                lambda x: x.requires_grad
+                if isinstance(x, torch.Tensor) else False,
+                detached_inputs,
+            ))
+        gradients = torch.autograd.grad(
+            outputs=temp_output,
+            inputs=filtered_detached_inputs + ctx.weights,
+            grad_outputs=grad_outputs,
+        )
+
+        input_gradients = []
+        i = 0
+        for rg in ctx.input_requires_grad:
+            if rg:
+                input_gradients.append(gradients[i])
+                i += 1
+            else:
+                input_gradients.append(None)
+
+        gradients = tuple(input_gradients) + gradients[-len(ctx.weights):]
+
+        return (None, None, None, None) + gradients
+
+
+class InvertibleModule(nn.Module, ABC):
+    r"""An abstract class for implementing Invertible Modules.
+
+    Args:
+        disable (bool, optional): This disables using the InvertibleFunction.
+            Therefore, it executes functions without memory savings.
+        num_bwd_passes (int, optional):
+            Number of backward passes to retain a link with the output.
+            After the last backward pass the output is discarded
+            and memory is freed. (default: :obj:`1`)
+    """
+    def __init__(self, disable=False, num_bwd_passes=1):
+        super().__init__()
+        self.disable = disable
+        self.num_bwd_passes = num_bwd_passes
+
+    def forward(self, *args):
+        y = self._fn_apply(args, self._forward, self._inverse)
+        return y
+
+    def inverse(self, *args):
+        x = self._fn_apply(args, self._inverse, self._forward)
+        return x
+
+    @abstractmethod
+    def _forward(self):
+        pass
+
+    @abstractmethod
+    def _inverse(self):
+        pass
+
+    def _fn_apply(self, args, fn, fn_inverse):
+        if not self.disable:
+            out = InvertibleFunction.apply(
+                fn,
+                fn_inverse,
+                self.num_bwd_passes,
+                len(args),
+                *(args + tuple(p
```

Comment:
This looks somewhat ugly. Does
```suggestion
                *args,
                *tuple(p for p in self.parameters() if p.requires_grad),
```
work as well?
work as well?